src/accounts.rs

```diff
@@ -81,23 +78,36 @@ impl AccountsDB {
         hash(&serialize(&ordered_accounts).unwrap())
     }
 
-    fn load(&self, pubkey: &Pubkey) -> Option<&Account> {
-        if let Some(account) = self.accounts.get(pubkey) {
-            return Some(account);
+    fn load<U>(checkpoints: &[U], pubkey: &Pubkey) -> Option<Account>
+    where
+        U: Deref<Target = Self>,
+    {
+        for db in checkpoints {
+            if let Some(account) = db.accounts.get(pubkey) {
+                return Some(account.clone());
+            }
         }
         None
     }
-
-    pub fn store(&mut self, pubkey: &Pubkey, account: &Account) {
+    /// Store the account update.  If the update is to delete the account because the token balance
+    /// is 0, purge needs to be set to true for the delete to occur in place.
+    pub fn store(&mut self, purge: bool, pubkey: &Pubkey, account: &Account) {
         if account.tokens == 0 {
-            self.accounts.remove(pubkey);
+            if purge {
```

Comment:
purge is hurting my brain...  is that only set on collapse of a fork?

Reply:
@rob-solana, yea when merging into root it’s set to true. I could probably hide it better.

Reply:
you could accept the same &[U] argument as load() and deduce purge by checkpoints.len()?